lib/segment/src/index/field_index/map_index/immutable_map_index.rs

```diff
@@ -43,13 +48,15 @@ impl<N: MapIndexKey + ?Sized> ImmutableMapIndex<N> {
     }
 
     /// Return mutable slice of a container which holds point_ids for given value.
+    ///
+    /// The returned slice is sorted and does contain deleted values.
     fn get_mut_point_ids_slice<'a>(
-        value_to_points: &mut HashMap<N::Owned, Range<u32>>,
+        value_to_points: &HashMap<N::Owned, (Range<u32>, u32)>,
```

Comment:
I would rather prefer to use bitmap, seems like more straightforward and already implemented in other places.